ban-cooking.lua

```diff
@@ -137,30 +137,29 @@ funcs.seeds = function()
         if p.material_defs.type.seed == -1 or p.material_defs.idx.seed == -1 or p.flags.TREE then goto continue end
         ban_cooking(p.name .. ' seeds', p.material_defs.type.seed, p.material_defs.idx.seed, df.item_type.SEEDS, -1)
         for _, m in ipairs(p.material) do
-            if m.id == "STRUCTURAL" and m.flags.EDIBLE_COOKED then
-                local has_drink = false
-                local has_seed = false
-                for _, s in ipairs(m.reaction_product.id) do
-                    has_seed = has_seed or s.value == "SEED_MAT"
-                    has_drink = has_drink or s.value == "DRINK_MAT"
-                end
-                if has_seed and has_drink then
-                    local matinfo = dfhack.matinfo.find(p.id, m.id)
-                    ban_cooking(p.name .. ' ' .. m.id, matinfo.type, matinfo.index, df.item_type.PLANT, -1)
+            if m.id == "STRUCTURAL" then
+                if m.flags.EDIBLE_COOKED then
+                    local has_seed = false
+                    for _, s in ipairs(m.reaction_product.id) do
+                        has_seed = has_seed or s.value == "SEED_MAT"
+                    end
+                    if has_seed then
+                        local matinfo = dfhack.matinfo.find(p.id, m.id)
+                        ban_cooking(p.name .. ' ' .. m.id, matinfo.type, matinfo.index, df.item_type.PLANT, -1)
+                    end
                 end
+                break
```

Comment:
this assumes that a plant cannot have more than one STRUCTURAL part. is this guaranteed, even in mods?

Reply:
Materials IDs (such as STRUCTURAL) have to be unique or items begin flip-flopping between being different materials on save and reload. If a modded plant does have more than one STRUCTURAL material, the save is going to have worse issues than ban-cooking not working on a plant.